### PR TITLE
Project Proposal FOCIL in Reth

### PR DESCRIPTION
**Pull request by PelleKrab**
Open to feedback :)

### CODE REVIEW

File: projects/focil-reth.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# Reth FOCIL Implementation
+
+## Motivation
+
+FOCIL (Fork-choice Enforced Inclusion Lists) [EIP-7805](https://eips.ethereum.org/EIPS/eip-7805) is a proposal that introduces Inclusion List (IL) committees for each block. These committees propose lists of transactions that must be included in the next block. It’s a response to the increasingly centralized nature of block building, currently dominated by a handful of builders.
+
+About 80% of blocks are built by just four builders, and approximately 3% of blocks actively censor transactions. This number has previously peaked at 85% and only dropped recently due to U.S. policy changes. But policies change, and centralization risks persist.
+
+FOCIL helps defend Ethereum's credible neutrality, which is a core value and major advantage over traditional financial systems and other blockchains (like Sui, which recently censored transactions to freeze a hack). Ethereum's trustless nature breaks down if transaction censorship becomes possible, at that point it starts to look more like a traditional payment network, heavily influenced by state policies. And thus just another traditional finance protocol, like Swift.
```
Comment (taxmeifyoucan) [inline projects/focil-reth.md:9]: The motivation part is good but I would keep it more simple, wouldn't mention other blockchains, just explain that censorship resistant is necessary for neutral, permissionless system. 

```suggestion
FOCIL helps defend Ethereum's credible neutrality, which is a core value and major advantage over traditional financial systems and other blockchains and provides censorship resistance. Ethereum's trustless nature breaks down if transaction censorship becomes possible, at that point it starts to look more like a traditional payment network, heavily influenced by state policies. And thus just another traditional finance protocol, like Swift.
```

File: projects/focil-reth.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+## Project Overview
+
+The goal of this project is to implement the FOCIL spec in the Reth execution client, enabling it to participate in interop testing with other clients. This involves updating and extending the current spec as outlined in [ethereum/execution-specs#1214](https://github.com/ethereum/execution-specs/pull/1214), and ensuring Reth plays well with other clients.
+
+The implementation builds on prior work by Jacob Kaufmann, rebasing and upgrading it to align with the latest specs and dependencies. A key focus is creating a clean, well thought out integration that not only supports interop testing but also helps advance the EIP across the broader Ethereum ecosystem.
```
Comment (taxmeifyoucan) [inline projects/focil-reth.md:15]: It would be good to link Jacob's branch here or later places where it's referenced

File: projects/focil-reth.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+- Unit tests for all new logic.
+- Simulate payload flow and validate Reth's behavior against the spec.
+- Integration tests for JSON-RPC endpoints and payload flows.
+- Ensure test coverage for edge cases and spec deviations.
+
+#### Extensive Interop Testing
+
+Engage in cross-client interoperability testing:
+
+- Use [Jihoon’s local devnet](https://github.com/jihoonsong/local-devnet-focil/tree/main) for simulations.
```
Comment (taxmeifyoucan) [inline projects/focil-reth.md:61]: The testing tool by Jihoon is using prysm, is it the only CL client with FOCIL prototype? An extension of the project could be contribute to this tool to add your reth prototype as an option. Potential other clients that have a working prototype, LH/Grandine would be a good rust pair to test with reth

**Review by taxmeifyoucan — APPROVED** [review]
Thanks Pelle, this looks good! 

One thing that comes to mind is the scope of the project, FOCIL is mainly a CL change. The proposal could explain a bit what architectural changes are needed in EL vs CL to give better overview of what is the overall scope of changes

File: projects/focil-reth.md (hunk 3 above)
Comment (PelleKrab) [reply projects/focil-reth.md:61]: I'm not too familiar with Kurtosis but it looks like there are other clients in the [config](https://github.com/jihoonsong/local-devnet-focil/blob/60524055c92b9ad35452310bf7ad82ec5301d93d/kurtosis/kurtosis.config.local.interop.yaml#L1-L19). But lemme add contributions this testing tool to the proposal.

**Comment by PelleKrab** [comment]
> Thanks Pelle, this looks good!
> 
> One thing that comes to mind is the scope of the project, FOCIL is mainly a CL change. The proposal could explain a bit what architectural changes are needed in EL vs CL to give better overview of what is the overall scope of changes

Would the bullets cover this or do you think it needs a more detailed description of CL vs EL FOCIL changes?

**Comment by taxmeifyoucan** [comment]
Sure bullet points, table or something simple is good enough. Maybe a short paragraph or so, it shouldn't be too long, just to demonstrate where is the burden of implementation
